diff --git a/dist/index.js b/dist/index.js
index e47582e39f60fcdf43152682758ac4ea177a48a1..dd1c8cea422dbdea515285775017ab6207895d0f 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -684,6 +684,9 @@ function convertToGoogleGenerativeAIMessages(prompt, options) {
             }
           }).filter((part) => part !== void 0)
         });
+        if (contents[contents.length - 1].parts.length === 0) {
+          contents.pop();
+        }
         break;
       }
       case "tool": {
diff --git a/dist/index.mjs b/dist/index.mjs
index 80cde81914e8c0a086fffff68d7c107966204537..203d145fc4af1ad9c8e67da45586ae155f438166 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -690,6 +690,9 @@ function convertToGoogleGenerativeAIMessages(prompt, options) {
             }
           }).filter((part) => part !== void 0)
         });
+        if (contents[contents.length - 1].parts.length === 0) {
+          contents.pop();
+        }
         break;
       }
       case "tool": {
diff --git a/dist/internal/index.js b/dist/internal/index.js
index 4eef7b1bfd407b433b2c3f8eef62eb65673f9c8d..f6adfcae3f18b039bd40f400a259223fe381fc87 100644
--- a/dist/internal/index.js
+++ b/dist/internal/index.js
@@ -468,6 +468,9 @@ function convertToGoogleGenerativeAIMessages(prompt, options) {
             }
           }).filter((part) => part !== void 0)
         });
+        if (contents[contents.length - 1].parts.length === 0) {
+          contents.pop();
+        }
         break;
       }
       case "tool": {
diff --git a/dist/internal/index.mjs b/dist/internal/index.mjs
index 23f58835df92cce6b52d8fa66d8a48e886f21efa..d91590124f05bfa1ce996198dc58bc2ce241b130 100644
--- a/dist/internal/index.mjs
+++ b/dist/internal/index.mjs
@@ -450,6 +450,9 @@ function convertToGoogleGenerativeAIMessages(prompt, options) {
             }
           }).filter((part) => part !== void 0)
         });
+        if (contents[contents.length - 1].parts.length === 0) {
+          contents.pop();
+        }
         break;
       }
       case "tool": {
